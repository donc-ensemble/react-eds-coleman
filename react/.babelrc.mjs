export default (api) => ({
  presets: [
    [
      "@babel/preset-react",
      {
        runtime: "automatic",
        useBuiltIns: true,
        useSpread: true,
        development: api.env("development"),
      },
    ],
    ["@babel/preset-typescript", {}],
  ],
  plugins: [],
});
