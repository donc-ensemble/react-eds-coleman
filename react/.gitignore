node_modules/

*storybook.log
coverage/
