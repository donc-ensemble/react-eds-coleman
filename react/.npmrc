# we don't the react package to end up in node_modules,
# when it's specified as a peerDependency of one of our dependencies, 
# since we're aliasing it to preact
omit=peer
