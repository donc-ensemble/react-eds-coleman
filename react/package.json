{
  "name": "eds-react-coleman",
  "version": "1.0.0",
  "type": "module",
  "engines": {
    "node": ">=20.0.0"
  },
  "scripts": {
    "webpack": "node --loader ts-node/esm -- node_modules/webpack/bin/webpack.js",
    "build:development": "cross-env NODE_ENV=development npm run webpack -- --config webpack.development.ts",
    "build:production": "cross-env NODE_ENV=production npm run webpack -- --config webpack.production.ts",
    "watch": "npm run build:development -- --watch",
    "test": "jest"
  },
  "author": "",
  "license": "ISC",
  "description": "",
  "dependencies": {
    "antd": "^5.22.0",
    "preact": "^10.24.3",
    "zustand": "^5.0.1"
  },
  "devDependencies": {
    "@babel/cli": "^7.25.9",
    "@babel/core": "^7.26.0",
    "@babel/preset-env": "^7.26.0",
    "@babel/preset-react": "^7.25.9",
    "@babel/preset-typescript": "^7.26.0",
    "@testing-library/dom": "^10.4.0",
    "@testing-library/jest-dom": "^6.6.3",
    "@testing-library/react": "^16.0.1",
    "@types/jest": "^29.5.14",
    "babel-loader": "^9.2.1",
    "clean-webpack-plugin": "^4.0.0",
    "cross-env": "^7.0.3",
    "css-loader": "^7.1.2",
    "identity-obj-proxy": "^3.0.0",
    "jest": "^29.7.0",
    "jest-environment-jsdom": "^29.7.0",
    "sass": "^1.80.6",
    "sass-loader": "^16.0.3",
    "style-loader": "^4.0.0",
    "ts-loader": "^9.5.1",
    "ts-node": "^10.9.2",
    "typescript": "^5.6.3",
    "webpack": "^5.96.1",
    "webpack-cli": "^5.1.4",
    "webpack-merge": "^6.0.1"
  }
}
