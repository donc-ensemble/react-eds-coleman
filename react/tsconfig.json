{
  "compilerOptions": {
    "paths": {
      "react/jsx-runtime": ["./node_modules/preact/jsx-runtime"],
      "react/jsx-dev-runtime": ["./node_modules/preact/jsx-runtime"],
      "react": ["./node_modules/preact/compat"],
      "react-dom/test-utils": ["./node_modules/preact/test-utils"],
      "react-dom/client": ["./node_modules/preact/compat/client"],
      "react-dom": ["./node_modules/preact/compat"],

      "@/*": ["src/*"]
    },
    "allowJs": true,
    "checkJs": false,
    "esModuleInterop": true,
    "forceConsistentCasingInFileNames": true,
    "jsx": "react-jsx",
    "lib": ["DOM", "DOM.Iterable", "ESNext"],
    "module": "ES2022",
    "moduleResolution": "Node",
    "outDir": "dist",
    "skipLibCheck": true,
    "sourceMap": true,
    "strict": true,
    "target": "ES6"
  }
}
