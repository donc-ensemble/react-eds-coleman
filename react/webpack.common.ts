import * as path from "path";
import * as fs from "fs";
import { Configuration } from "webpack";

const blocksSrcPath = path.resolve(import.meta.dirname, "src/blocks");
const blocks = fs
  .readdirSync(blocksSrcPath, { withFileTypes: true })
  .filter((dirent) => dirent.isDirectory());

const configuration: Configuration = {
  entry: {
    ...blocks
      .map((block) => ({
        [`blocks:${block.name}`]: {
          filename: `blocks/${block.name}/${block.name}.js`,
          import: path.resolve(blocksSrcPath, block.name, `${block.name}.tsx`),
          dependOn: ["bundle"],
          library: {
            type: "module",
          },
        },
      }))
      .reduce((blocks, block) => ({ ...blocks, ...block }), {}),
    delayed: {
      filename: "scripts/delayed.js",
      import: ["../scripts/_delayed.js"],
      dependOn: ["bundle"],
      library: {
        type: "module",
      },
    },
    scripts: {
      filename: "scripts/scripts.js",
      import: ["../scripts/_scripts.js"],
      dependOn: ["bundle"],
      library: {
        type: "module",
      },
    },
    bundle: {
      filename: "scripts/bundle.js",
      import: ["../scripts/aem.js"],
      library: {
        type: "module",
      },
    },
  },

  experiments: {
    outputModule: true,
  },

  output: {
    path: path.resolve(import.meta.dirname, ".."),
    publicPath: "/",
  },

  optimization: {
    splitChunks: {
      cacheGroups: {
        bundle: {
          test: /[\\/]node_modules[\\/]|aem\.js$/,
          chunks: "all",
          name: "bundle",
          enforce: true,
        },
      },
    },
  },

  resolve: {
    alias: {
      "react/jsx-runtime": "preact/jsx-runtime",
      "react/jsx-dev-runtime": "preact/jsx-runtime",
      react: "preact/compat",
      "react-dom/test-utils": "preact/test-utils",
      "react-dom": "preact/compat", // Must be below test-utils

      "@": path.resolve(import.meta.dirname, "src"),
    },
    extensions: [".mjs", ".js", ".jsx", ".mts", ".ts", ".tsx"],
    plugins: [],
  },

  module: {
    rules: [
      {
        test: /\.(ts|tsx)$/,
        exclude: /node_modules/,
        use: "babel-loader",
      },
      {
        test: /\.css$/,
        use: ["style-loader", "css-loader"],
      },
      {
        test: /\.s[ac]ss$/,
        use: ["style-loader", "css-loader", "sass-loader"],
      },
    ],
  },
  plugins: [],
  target: ["es2022"],
};

export default configuration;
