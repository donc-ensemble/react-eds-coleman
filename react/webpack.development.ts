import { merge } from "webpack-merge";
import common from "./webpack.common.js";
import webpack from "webpack";

export default merge(common, {
  mode: "development",
  plugins: [
    new webpack.EnvironmentPlugin({
      NODE_ENV: "development",
    }),
  ],
  devtool: "inline-source-map",
});
