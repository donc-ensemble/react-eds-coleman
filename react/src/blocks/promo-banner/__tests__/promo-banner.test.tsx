import "@testing-library/jest-dom";
import { render, screen } from "@testing-library/react";
import {
  PromoBanner,
  PromoBannerProps,
} from "@/blocks/promo-banner/promo-banner";
import { describe, it } from "@jest/globals";

describe("Promo Banner Component", function () {
  it("should render promo banner with picture sources and texts", function () {
    // Arrange
    const promoBannerProps: PromoBannerProps = {
      pictureSources: [
        {
          type: "image",
          src: "https://example.com/image.jpg",
        },
      ],
      texts: ["Text 1", "Text 2"],
    };
    render(<PromoBanner {...promoBannerProps} />);

    // Act

    // Assert
    const paragraphs = screen.getAllByRole("paragraph");
    expect(paragraphs).toHaveLength(2);
    paragraphs.forEach((paragraph, index) => {
      expect(paragraph).toHaveTextContent(["Text 1", "Text 2"][index]);
    });
  });

  it("rendered promo banner matches snapshot", function () {
    // Arrange
    const promoBannerProps: PromoBannerProps = {
      pictureSources: [
        {
          type: "image",
          src: "https://example.com/image.jpg",
        },
      ],
      texts: ["Snapshot text 1", "Snapshot text 2"],
    };
    const { asFragment } = render(<PromoBanner {...promoBannerProps} />);

    // Act

    // Assert
    expect(asFragment()).toMatchSnapshot();
  });
});
