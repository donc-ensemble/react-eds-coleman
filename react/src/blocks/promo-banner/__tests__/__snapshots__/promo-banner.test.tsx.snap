// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Promo Banner Component rendered promo banner matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="wrapper"
  >
    <div>
      <picture>
        <img
          alt=""
          height="333"
          loading="lazy"
          src="https://example.com/image.jpg"
          width="1600"
        />
      </picture>
    </div>
    <div>
      <p>
        Snapshot text 1
      </p>
      <p>
        Snapshot text 2
      </p>
    </div>
  </div>
</DocumentFragment>
`;
